feat: add periodic push to svc-gis

amsmith-pro opened the pull request.

amsmith-pro committed:
feat: add periodic push to svc-gis

owlot commented on server/src/config.rs:
```diff
@@ -31,6 +44,10 @@ impl Config {
         log::warn!("Creating Config object with default values.");
         Config {
             docker_port_grpc: 50051,
+            gis_host_grpc: String::from("svc-gis"),
```
These new fields should be added to the tests below as well

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+        .collect();
+
+    if nodes.is_empty() {
+        grpc_warn!("(waypoints_loop) no waypoints to update.");
```
wrong function name in debug message:
(waypoints_loop) -> (update_waypoints)

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+    let address = format!("{}:{}", host, port);
+    let result = GisClient::connect(address).await;
+    let Ok(mut client) = result else {
+        grpc_error!("Failed to connect to gRPC server: {}", result.unwrap_err());
```
function name missing in debug message

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+
+    match client.update_waypoints(request).await {
+        Ok(response) => {
+            grpc_info!("RESPONSE={:?}", response);
```
Should this really be an info message? Feels more like a debug message
Also add the function name prefix

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+            grpc_info!("RESPONSE={:?}", response);
+        }
+        Err(e) => {
+            grpc_error!("ERROR={:?}", e);
```
Function name missing in debug message

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+            Some(t) => match datetime_to_timestamp(&t) {
+                Some(t) => Some(t),
+                _ => {
+                    grpc_error!("(restrictions_loop) could not convert timestamp for zone with label {label}.");
```
Wrong function name in log message -> (update_restrictions)

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+            Some(t) => match datetime_to_timestamp(&t) {
+                Some(t) => Some(t),
+                _ => {
+                    grpc_error!("(restrictions_loop) could not convert timestamp for zone with label {label}.");
```
Wrong function name in log message -> (update_restrictions)

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+    }
+
+    if zones.is_empty() {
+        grpc_warn!("(restrictions_loop) no restrictions to update.");
```
Wrong function name in log message -> (update_restrictions)

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+    let address = format!("{}:{}", host, port);
+    let result = GisClient::connect(address).await;
+    let Ok(mut client) = result else {
+        grpc_error!("Failed to connect to gRPC server: {}", result.unwrap_err());
```
Missing function name in log message

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+
+    match client.update_no_fly_zones(request).await {
+        Ok(response) => {
+            grpc_info!("RESPONSE={:?}", response);
```
I think this should be a debug instead again
Also missing function name prefix

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+            grpc_info!("RESPONSE={:?}", response);
+        }
+        Err(e) => {
+            grpc_error!("ERROR={:?}", e);
```
Missing function name prefix

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot commented on server/src/grpc/server.rs:
```diff
@@ -357,58 +450,4 @@ mod tests {
         println!("{:?}", result);
         assert_eq!(result.released, true);
     }
-
```
New tests are missing for the newly added functions

amsmith-pro replied:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

owlot reviewed (commented):
I've marked mostly log message issues.
And I think some tests can be added.

amsmith-pro committed:
chore: address reviewer comments

amsmith-pro commented:
@owlot Updated log functions, added tests - 

Switched to the newer way of instantiating grpc clients so that it could be unit tested, works great!

Also changed the behavior of when timestamps can't be converted (which shouldn't happen, but jic) in `update_restrictions` to just set start/end time to None (start immediately/end never) rather than discarding the restriction, which could have some hairy real-life consequences
